pydantic_ai_slim/pydantic_ai/agent/__init__.py

```diff
@@ -286,6 +290,9 @@ def __init__(
                 Each processor takes a list of messages and returns a modified list of messages.
                 Processors can be sync or async and are applied in sequence.
             event_stream_handler: Optional handler for events from the model's streaming response and the agent's execution of tools.
+            tool_timeout: Default timeout in seconds for tool execution. If a tool takes longer than this,
+                a retry prompt is returned to the model. Individual tools can override this with their own timeout.
```

Comment:
Let's make it clear that "the tool is considered to have failed" in terms of counting towards the retry limit.